openbas-api/src/main/java/io/openbas/rest/health_check/HealthCheckApi.java

```diff
@@ -0,0 +1,65 @@
+package io.openbas.rest.health_check;
+
+import io.openbas.rest.helper.RestBehavior;
+import io.openbas.service.HealthCheckService;
+import io.openbas.service.exception.HealthCheckFailureException;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+@RestController
+public class HealthCheckApi extends RestBehavior {
+
+  public static final String HEALTH_CHECK_URI = "/api/health";
+
+  private HealthCheckService healthCheckService;
+
+  private String healthCheckKey;
+
+  @Autowired
+  public void setHealthCheckService(HealthCheckService healthCheckService) {
+    this.healthCheckService = healthCheckService;
+  }
+
+  @Autowired
+  public void setHealthCheckKey(
+      @Value("${openbas.healthcheck.key:#{null}}") String healthCheckKey) {
+    this.healthCheckKey = healthCheckKey;
+  }
+
+  @GetMapping(HEALTH_CHECK_URI)
+  @Operation(
+      summary = "Run an healthcheck ",
+      description = "Tries to connect to dependencies (DB/Minio/RabbitMQ)")
+  @ApiResponses(
+      value = {
+        @ApiResponse(responseCode = "200", description = "Service is healthy"),
+        @ApiResponse(responseCode = "503", description = "Service is not running properly")
```

Comment:
I see you make use of constants further in the code, would you consider using constants here too?

Reply:
nitpick: Additionally, would you consider tying the documentation to the actual code with an in-class constant? i.e.
```
private static final HttpStatus ALL_GOOD_STATUS = HttpStatus.OK;
private static final HttpStatus BROKEN_STATUS = HttpStatus.SERVICE_UNAVAILABLE;
...
@ApiResponse(responseCode = ALL_GOOD_STATUS.value(), description = "Service is healthy"),
@ApiResponse(responseCode = BROKEN_STATUS.value(), description = "Service is not running properly")
...
    throw new ResponseStatusException(
          HttpStatusCode.valueOf(BROKEN_STATUS.value()), message);
    }
    return new ResponseEntity<>("success", ALL_GOOD_STATUS);
```

Reply:
Yes, that makes sense I will update it

Reply:
I would need to add a constant string so it doesn't make too much sense or there is another way but I didn't spent too much time thinking about that